test/RewardPoolV2.js

```diff
@@ -179,13 +247,260 @@ describe("RewardPool", function () {
+      await stAzur.connect(user).depositFor(user3, DEPOSIT);
+
+      await stAzur.connect(owner).updateStakingIncentive(INCENTIVE_REWARD, INCENTIVE_DURATION);
+      expect(await azur.balanceOf(owner.address)).to.be.equal(ownerBalanceBefore - INCENTIVE_REWARD);
+
+      await timeShiftBy(ethers, INCENTIVE_DURATION);
+
+      for (const account of [user, user2, user3]) {
+        expect(await calculateAzurBalance(account.address)).to.be.closeToRelative(
+            DEPOSIT + INCENTIVE_REWARD / 3n,
+        );
+      }
+    });
+    it("Stake before the start of token distribution and unstake it a long time ago of the end", async function () {
+      await stAzur.connect(user).depositFor(user, DEPOSIT);
+      await stAzur.connect(owner).updateStakingIncentive(INCENTIVE_REWARD, INCENTIVE_DURATION);
+      await timeShiftBy(ethers, ONE_YEAR * 5);
+
+      expect(await calculateAzurBalance(user.address)).to.be.closeToRelative(
+          DEPOSIT + INCENTIVE_REWARD
+      );
+    });
+    it("Stake before the start of token distribution and claim reward before it ends", async function () {
+      for (const account of [user, user2, user3]) {
+        await stAzur.connect(account).depositFor(account, DEPOSIT);
+      }
+
+      const distributionStartedAt = await updateStakingIncentive(INCENTIVE_REWARD, INCENTIVE_DURATION);
+
+      await timeShift(distributionStartedAt + INCENTIVE_DURATION / 3 - 1);
+
+      const res1 = await requestWithdrawal(user, DEPOSIT);
+      expect(res1.withdrawalAmount).to.be.closeToRelative(DEPOSIT + INCENTIVE_REWARD / 9n);
+
+      await timeShift(distributionStartedAt + (INCENTIVE_DURATION * 2) / 3 - 1);
+      const res2 = await requestWithdrawal(user2, DEPOSIT);
+      expect(res2.withdrawalAmount).to.be.closeToRelative(
+          DEPOSIT + (INCENTIVE_REWARD * 5n) / 18n,
+      );
+
+      await timeShift(distributionStartedAt + INCENTIVE_DURATION);
+      const res3 = await requestWithdrawal(user3, DEPOSIT);
+      expect(res3.withdrawalAmount).to.be.closeToRelative(
+          DEPOSIT + (INCENTIVE_REWARD * 11n) / 18n,
+      );
+
+      for (const account of [user, user2, user3]) {
+        await expect(requestWithdrawal(account, 1)).to.be.revertedWithCustomError(stAzur, "ERC20InsufficientBalance");
+      }
+    });
+    it.only("Stake after the start of token distribution and claim reward after it ends", async function () {
+      await stAzur.connect(user).depositFor(user, DEPOSIT);
+
+      const distributionStartedAt = await updateStakingIncentive(INCENTIVE_REWARD, INCENTIVE_DURATION);
+
+      await timeShiftBy(ethers,distributionStartedAt + INCENTIVE_DURATION / 3 - 1);
+      await stAzur.connect(user).depositFor(user2, DEPOSIT);
+
+      await timeShiftBy(ethers,distributionStartedAt + (INCENTIVE_DURATION * 2) / 3 - 1);
+      await stAzur.connect(user).depositFor(user3, DEPOSIT);
+
+      await timeShiftBy(ethers,distributionStartedAt + INCENTIVE_DURATION);
+
+      await unstake(user, stakeId);
```

Comment:
error at test run:
`ReferenceError: unstake is not defined at Context.<anonymous> (test/RewardPoolV2.js:317:7)`